src/mixing_fab_tests.cc

```diff
@@ -0,0 +1,247 @@
+#include "mixing_fab_tests.h"
+
+
+#include <gtest/gtest.h>
+#include "equality_helpers.h"
+#include "cyclus.h"
+
+using pyne::nucname::id;
+using cyclus::Composition;
+using cyclus::CompMap;
+using cyclus::compmath::Normalize;
+using cyclus::compmath::Add;
+
+using cyclus::Material;
+using cyclus::QueryResult;
+using cyclus::Cond;
+using cyclus::toolkit::MatQuery;
+
+namespace cycamore {
+  
+  
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  
+  Composition::Ptr c_pustream() {
+    CompMap m;
+    m[id("pu239")] = 100;
+    m[id("pu240")] = 10;
+    m[id("pu241")] = 1;
+    m[id("pu242")] = 1;
+    return Composition::CreateFromMass(m);
+  };
+  
+  Composition::Ptr c_natu() {
+    CompMap m;
+    m[id("u235")] =  .007;
+    m[id("u238")] =  .993;
+    return Composition::CreateFromMass(m);
+  };
+  
+  Composition::Ptr c_uox() {
+    CompMap m;
+    m[id("u235")] = 0.04;
+    m[id("u238")] = 0.96;
+    return Composition::CreateFromMass(m);
+  };
+
+  void MixingFabTest::SetUp() {
+    mf_facility_ = new MixingFab(tc_.get());
+    InitParameters();
+    SetUpMixingFab();
+  }
+  
+  void MixingFabTest::TearDown() {
+    delete mf_facility_;
+  }
+  
+  void MixingFabTest::InitParameters(){
+    in_com.push_back("in_c1"); in_frac.push_back(0.80); in_cap.push_back(30);
+    in_com.push_back("in_c2"); in_frac.push_back(0.15); in_cap.push_back(20);
+    in_com.push_back("in_c3"); in_frac.push_back(0.05); in_cap.push_back(10);
+    
+    out_com = "out_com"; out_cap = 50;
+    
+    throughput = 20;
+  }
+  
+  void MixingFabTest::SetUpMixingFab(){
+    mf_facility_->in_commods = in_com;
+    mf_facility_->mixing_frac = in_frac;
+    mf_facility_->in_buf_size = in_cap;
+    
+    mf_facility_->out_commod = out_com;
+    mf_facility_->output_buf_size = out_cap;
+    
+    mf_facility_->throughput = throughput;
+  }
+  
+  void MixingFabTest::TestInitState(MixingFab* fac){
```

Comment:
These functions' contents should just be in the TEST macro.  The indirection just obfuscates things.